### PR TITLE
Skip uninstall in ci if manual

### LINKED ISSUE #59835: CI: don’t run `uninstall` in `installer manual:` casks
`installer manual:` casks can’t install in CI (because there is no manual intervention), so their `uninstall` will fail, which means `installer manual:` casks will always fail CI. We should make an exception for those casks in CI, to run everything *except* `uninstall`.

### PR DESCRIPTION
**Pull request by hyuraku**
Closes https://github.com/Homebrew/homebrew-cask/issues/59835

### CODE REVIEW

File: cmd/brewcask-ci.rb [hunk 1]
```diff
@@ -174,6 +174,10 @@ def modified_cask_files
       def added_cask_files
         @added_cask_files ||= added_files.select { |path| tap.cask_file?(path) }
       end
+
+      def check_manual
+        cask.ancestors.include?(ManualInstaller)
```
Comment (reitermarkus) [inline cmd/brewcask-ci.rb]:
```suggestion
        cask.artifacts.any? { |a| a.is_a?(Artifact::Installer::ManualInstaller) }
```

File: cmd/brewcask-ci.rb [hunk 2]
```diff
@@ -174,6 +174,10 @@ def modified_cask_files
       def added_cask_files
         @added_cask_files ||= added_files.select { |path| tap.cask_file?(path) }
       end
+
+      def check_manual
```
Comment (reitermarkus) [inline cmd/brewcask-ci.rb]:
```suggestion
      def manual_installer?
```

**Review by reitermarkus — CHANGES_REQUESTED** [review]

**Comment by reitermarkus** [comment]
Can you add a commit with a change to a cask this applies to, to see if this is working?

File: Casks/blue-jeans.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 cask 'blue-jeans' do
-  version '2.14.0.487'
-  sha256 'bb55e0c2c65d1ed748c1db3c7e13255e57b9f29930da4b2a7f8d1f164791fe11'
+  version '2.13.1.17'
+  sha256 'e7a5f5faa69000906bb30da34edc415d297174b4c463dd186c00125be58140c9'
```
Comment (hyuraku) [inline Casks/blue-jeans.rb]: To check fixed ci, I changed one cask.
I will reset it after fixed ci is approved.

File: Casks/blue-jeans.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
...
 
   url "https://swdl.bluejeans.com/desktop-app/mac/#{version.major_minor_patch}/#{version}/BlueJeansInstaller.dmg"
   appcast 'https://www.bluejeans.com/downloads'
```
Comment (vitorgalvao) [inline Casks/blue-jeans.rb]: Also remove this line on the test, or CI will fail because of it.

File: cmd/brewcask-ci.rb [hunk 5]
```diff
@@ -64,7 +64,7 @@ def run
 
           overall_success &= step "brew cask uninstall #{cask.token}", "uninstall" do
             success = begin
-              Installer.new(cask, verbose: true).uninstall
+              Installer.new(cask, verbose: true).uninstall unless manual_installer?(cask)
```
Comment (reitermarkus) [inline cmd/brewcask-ci.rb]: Also, let's add a message like `"Cask has a manual installer, skipping …"`.

**Comment by reitermarkus** [comment]
Thanks, @hyuraku, working as expected! You can remove the change to the cask again.

File: cmd/brewcask-ci.rb [hunk 6]
```diff
@@ -174,6 +174,14 @@ def modified_cask_files
       def added_cask_files
         @added_cask_files ||= added_files.select { |path| tap.cask_file?(path) }
       end
+
+      def manual_installer?(cask)
+        if cask.artifacts.any? { |artifact| artifact.is_a?(Artifact::Installer::ManualInstaller) }
+          puts 'Cask has a manual installer, skipping …'
```
Comment (reitermarkus) [inline cmd/brewcask-ci.rb]: Move the message to where `manual_installer?` is used, so this function can stay simple.

**Review by vitorgalvao — APPROVED** [review]

**Comment by reitermarkus** [comment]
Great work, @hyuraku!

**Pull request merged**